### DISCUSSION TITLE
How to redirect 404's from the proxied "back" site back to the "front" site?

### DISCUSSION THREAD

**Discussion by jakenuts** [dotnet/yarp#1949] [Q&A] [answered]
I'm using Yarp to implement a migration from ASP.NET 4 to dotnet 7.0 with the systemnet adapters. It all seems to be working well but when a page is requested and both the new site and old site don't have a matching page I'd like the request redirected to a /NotFound route on the new site. Puzzled over the configuration docs for a bit and am not entirely sure how one might accomplish that. 

Thanks!

James

**Comment by samsp-msft** [comment, marked as answer]
If you have a catchall route with a high order property (higher number is lower priority) then that should get called only if something more specific doesn't match first.

**Reply by Tratcher** [reply]
That doesn't handle 404s from the backend though. A response transform could do it.
```C#
            services.AddReverseProxy()
                .LoadFromConfig(_configuration.GetSection("ReverseProxy"))                                       
                .AddTransforms(transformBuilderContext =>  // Add transforms inline
                {
                    transformBuilderContext.AddResponseTransform(transformContext =>
                    {
                        if (transformContext.HttpContext.Response.StatusCode == StatusCodes.Status404NotFound)
                        {
                            transformContext.HttpContext.Response.Redirect("/NotFound");
                            transformContext.SuppressResponseBody = true;
                        }
                        return default;
                    });
                });
```

**Reply by jakenuts** [reply]
Thanks! Glad to know I didn't miss anything obvious in the documentation on appsettings... 8)

**Reply by jakenuts** [reply]
Worked a charm, thanks again!
